sdks/typescript/src/apache_beam/transforms/transform.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as runnerApi from "../proto/beam_runner_api";
+import { PValue } from "../pvalue";
+import { Pipeline } from "../internal/pipeline";
+
+export function withName<T>(name: string | (() => string), arg: T): T {
+  (arg as any).beamName = name;
+  return arg;
+}
+
+export function extractName<T>(withName: T): string {
+  const untyped = withName as any;
+  if (untyped.beamName != undefined) {
+    if (typeof untyped.beamName == "string") {
+      return untyped.beamName;
+    } else {
+      return untyped.beamName();
+    }
+  } else if (
+    untyped.name != undefined &&
+    untyped.name &&
```

Comment:
```suggestion
    untyped.name &&
```

The first check here is superfluous since if `untyped.name` is undefined, `untyped.name` will evaluate to false